src/index.ts

```diff
@@ -46,18 +52,37 @@ async function main() {
         },
     );
 
-    // Create Firefly client
-    const fireflyClient = new FireflyClient(logger, accessKey, secretKey);
-
     // Handle tool requests
     server.setRequestHandler(
         CallToolRequestSchema,
-        async (request: CallToolRequest) => {
+        async (request: CallToolRequest, extra: RequestHandlerExtra) => {
             logger.error("Received CallToolRequest:", request);
             try {
                 const toolName = request.params.name;
                 const args = request.params.arguments || {};
 
+                let fireflyClient: FireflyClient | null = null;
+                if (!sse) {
+                    fireflyClient = stdioFireflyClient;
+                }
+
+                if (extra.sessionId) {
```

Comment:
is `extra.sessionId` a generated random string? can someone hijack a session by just trying different ids (in case it's incremental)?